src/book/test/integration/service.int-spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Test } from '@nestjs/testing';
+import { Author, BookStatus, Category } from '@prisma/client';
+import { AppModule } from 'src/app.module';
+import { BookDTO } from 'src/book/dtos';
+import { BooksService } from 'src/book/service';
+import { PrismaService } from 'src/prisma/service';
+
+describe('BookServices Int', () => {
+  let prisma: PrismaService;
+  let bookServices: BooksService;
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    prisma = moduleRef.get(PrismaService);
+    bookServices = moduleRef.get(BooksService);
+    await prisma.cleanDatabase();
+  });
+  let bookId: string;
+  const dto: BookDTO = {
+    title: 'test book',
+    coverImage: 'test cover image',
+    description: 'some description',
+    language: 'vi-VN',
+    numberOfPages: 999,
+    publishDate: new Date(),
+    publisher: 'IV',
+    reprintTimes: 7,
+  };
+  describe('create book', () => {
+    let authorId: string;
+    let categoryId: string;
+    it('should create author', async () => {
+      const author = await prisma.author.create({
```

Comment:
This is not correct for testing.
When you want to test whether author is created on book creation, you should check how many time the prisma.author.create has been called. You can check the spy function of jest for this.